sdk/core/azure_core/src/http/options/mod.rs

```diff
@@ -27,6 +29,17 @@ pub struct ClientOptions {
 
     /// User-Agent telemetry options.
     pub user_agent: Option<UserAgentOptions>,
+
+    /// Options for request instrumentation, such as distributed tracing.
+    ///
+    /// If not specified, defaults to no instrumentation.
+    ///
+    pub request_instrumentation: Option<RequestInstrumentationOptions>,
```

Comment:
Nit: isn't this request and response instrumentation, though? Could we just call this "instrumentation"?

Reply:
Good point - technically it's request instrumentation and public API instrumentation, but it eventually could grow to include metrics and other constructs.